src/cli/index.js

```diff
@@ -0,0 +1,231 @@
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+
+var packageInstall = require('../meta/package-install');
+var dirname = require('./paths').baseDir;
+
+// check to make sure dependencies are installed
+try {
+	fs.readFileSync(path.join(dirname, 'package.json'));
+	fs.readFileSync(path.join(dirname, 'node_modules/async/package.json'));
+} catch (e) {
+	if (e.code === 'ENOENT') {
+		process.stdout.write('Dependencies not yet installed.\n');
+		process.stdout.write('Installing them now...\n\n');
+
+		packageInstall.updatePackageFile();
+		packageInstall.preserveExtraneousPlugins();
+		packageInstall.npmInstallProduction();
+
+		require('colors');
+		process.stdout.write('OK'.green + '\n'.reset);
+	} else {
+		throw e;
+	}
+}
+
+require('colors');
+var nconf = require('nconf');
+var program = require('commander');
+
+var pkg = require('../../package.json');
+var file = require('../file');
+var prestart = require('../prestart');
+
+program
+	.name('./nodebb')
+	.description('Welcome to NodeBB')
+	.version(pkg.version)
+	.option('--json-logging', 'Output to logs in JSON format', false)
+	.option('--log-level <level>', 'Default logging level to use', 'info')
+	.option('-d, --dev', 'Development mode, including verbose logging', false)
+	.option('-l, --log', 'Log subprocess output to console', false)
+	.option('-c, --config <value>', 'Specify a config file', 'config.json')
```

Comment:
`--no-silent` and `--no-daemon`?

Reply:
No that's fine, if they aren't passed through right now, no need to do it :smile: